src/components/fields/ArrayField.js

```diff
@@ -312,14 +312,24 @@ class ArrayField extends Component {
     const arrayProps = {
       canAdd: this.canAddItem(formData),
       items: formData.map((item, index) => {
+        const itemSchema = retrieveSchema(
+          schema.items,
+          definitions,
+          formData[index]
```

Comment:
Good point, I felt like I was "hacking" a bit as I am still familiarizing myself with the code base. Thank you for the feedback. I will update.